consensus/src/tests/consensus_tests.rs

```diff
@@ -163,19 +164,19 @@ async fn dead_node() {
         }
     });
 
-    // We should commit 4 leaders (rounds 1, 3, 5, and 7).
-    for i in 1..=24 {
+    // We should commit 4 leaders (rounds 2, 4, 6, and 8).
+    for i in 1..=21 {
         let certificate = rx_output.recv().await.unwrap();
         let expected = ((i - 1) / keys.len() as u64) + 1;
-        //println!("{}, {}, {}", certificate.round(), expected, keys.len());
+        println!("{}, {}, {}", certificate.round(), expected, keys.len());
         assert_eq!(certificate.round(), expected);
     }
     let certificate = rx_output.recv().await.unwrap();
-    assert_eq!(certificate.round(), 7);
+    assert_eq!(certificate.round(), 8);
 }
 
-// Run for 5 dag rounds. The leaders of round 2 does not have enough support, but the leader of
-// round 4 does. The leader of rounds 2 and 4 should thus be committed.
+// Run for 7 dag rounds. The leaders of round 2 does not have enough support (f+1 votes), but the leader of
```

Comment:
Isn't the threshold back to 2f+1 now that we have a fallback?

Reply:
Yes the threshold is 2f+1 but once the fallback leader in fallback wave 1 (leader of round 4) is committed we must check whether earlier leaders may have been committed. We thus commit the steady state leader of round 2 (steady state wave 1) since there are f+1 votes for it and less than f+1 votes for the fallback leader of fallback wave 0.